File: rnapp/ios/FlutterModule.swift
import Flutter
import React

@objc(FlutterModule)
class FlutterModule: NSObject {
    private var methodChannel: FlutterMethodChannel?
    let flutterDependencies: FlutterDependencies
    
    override init() {
        flutterDependencies = FlutterCacheManager.shared.getOrCreateDependency()
        super.init()
        initializeMethodChannel()
    }
    
    @objc
    func showProfile(_ userId: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        print("showProfile called with userId: \(userId)")
        invokeFlutterMethod("showProfile", arguments: userId, resolver: resolver, rejecter: rejecter)
    }
    
    @objc
    func callFlutterMethod(_ method: String, arguments: Any, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        print("callFlutterMethod: \(method) with arguments: \(arguments)")
        invokeFlutterMethod(method, arguments: arguments, resolver: resolver, rejecter: rejecter)
    }
    
    private func invokeFlutterMethod(_ method: String, arguments: Any, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async { [weak self] in
            guard let self = self else {
                rejecter("error", "Module instance is nil", nil)
                return
            }
            
            if self.methodChannel == nil {
                self.initializeMethodChannel()
            }
            
            guard let methodChannel = self.methodChannel else {
                rejecter("error", "Failed to initialize method channel", nil)
                return
            }
            
            print("Invoking Flutter method: \(method) with arguments: \(arguments)")
            methodChannel.invokeMethod(method, arguments: arguments) { (response) in
                if let error = response as? FlutterError {
                    print("Flutter method error: \(error.message ?? "Unknown error")")
                    rejecter("error", error.message, nil)
                } else if let result = response {
                    print("Flutter method success with result: \(result)")
                    resolver(result)
                } else {
                    print("Flutter method completed with no result")
                    resolver(NSNull())
                }
            }
        }
    }
    
    private func initializeMethodChannel() {
        print("Initializing method channel")
        methodChannel = FlutterCacheManager.shared.getOrCreateMethodChannel(for: flutterDependencies.flutterEngine)
    }
}

File: rnapp/android/app/src/main/java/com/rnapp/FlutterModule.kt
package com.rnapp

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import io.flutter.plugin.common.MethodChannel

class FlutterModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    private var methodChannel: MethodChannel? = null

    init {
        initializeMethodChannel()
    }

    private fun initializeMethodChannel() {
        methodChannel = FlutterCacheManager.getMethodChannel()
    }

    @ReactMethod
    fun showProfile(userId: String, promise: Promise) {
        invokeFlutterMethod("showProfile", userId, promise)
    }

    @ReactMethod
    fun callFlutterMethod(method: String, arguments: String?, promise: Promise) {
        invokeFlutterMethod(method, arguments, promise)
    }

    private fun invokeFlutterMethod(method: String, arguments: Any?, promise: Promise) {
        if (methodChannel == null) {
            initializeMethodChannel()
        }

        methodChannel?.let {
            currentActivity?.runOnUiThread {
                try {
                    println("FlutterModule: $method called with arguments: $arguments")
                    it.invokeMethod(method, arguments)
                    promise.resolve(null)
                } catch (e: Exception) {
                    promise.reject("ERROR", "Failed to invoke Flutter method: ${e.message}")
                }
            }
        } ?: run {
            promise.reject("ERROR", "Method channel not initialized")
        }
    }

    override fun getName(): String {
        return "FlutterModule"
    }
}